src/enclave_proc/resource_manager.rs

```diff
@@ -469,13 +470,26 @@ impl EnclaveHandle {
         debug_mode: bool,
     ) -> NitroCliResult<Self> {
         let requested_mem = memory_mib << 20;
+        let link_path = read_link(PathBuf::from(format!(
```

Comment:
We're introducing extra error cases here which don't make sense I think. The metadata call can only likely only fail with EBADFD or some kernel error. read_link can fail in a thousand different ways though, and is not required. The failures are not necessarily related to the file itself, but tree traversal and other nonsence. I really don't think we need to do this here, as it's super error prone and doesn't improve logging IMO.

Reply:
Right, some of the File errors are not related directly to files stored on disk, but rather to the descriptor obtained after opening a file. In this case, `FileOperationFailures` would only correspond to actions failing because a file is not found or permissions don't allow for opening it etc.
Although the changes above deal with descriptors, I would not prefer to introduce an additional error code for something as `DescriptorOperationFailure`, since I don't think it really helps customers debug the problem themselves. Or should I consider adding this separate error code and use it above?

Reply:
I understand that `read_link` brings in extra complexity and possibilities for failures, but there's no other way of retrieving a file path starting from the fd, that I am aware of.

Reply:
Removed the file path from this section too. Indeed reporting the file path here would not help that much. And, since the error comes from an file descriptor operation for which `read_link` will most likely fail too.